src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/ParamservUtils.java

```diff
@@ -86,14 +98,44 @@ public static void cleanupData(Data data) {
 	 * @return new sliced matrix
 	 */
 	public static MatrixObject sliceMatrix(MatrixObject mo, long rl, long rh) {
-		MatrixObject result = new MatrixObject(Expression.ValueType.DOUBLE, null,
-			new MetaDataFormat(new MatrixCharacteristics(-1, -1, -1, -1),
-				OutputInfo.BinaryBlockOutputInfo, InputInfo.BinaryBlockInputInfo));
+		MatrixObject result = newMatrixObject();
 		MatrixBlock tmp = mo.acquireRead();
 		result.acquireModify(tmp.slice((int) rl - 1, (int) rh - 1, 0, tmp.getNumColumns() - 1, new MatrixBlock()));
 		mo.release();
 		result.release();
 		result.enableCleanup(false);
 		return result;
 	}
+
+	public static MatrixObject generatePermutation(MatrixObject mo, ExecutionContext ec) {
+		// Create the sequence
+		double[] data = LongStream.range(1, mo.getNumRows() + 1).mapToDouble(l -> l).toArray();
+		MatrixBlock seqMB = DataConverter.convertToMatrixBlock(data, true);
+		MatrixObject seq = ParamservUtils.newMatrixObject();
+		seq.acquireModify(seqMB);
+		seq.release();
+		ec.setVariable("seq", seq);
+
+		// Generate a sample
+		DataGenCPInstruction sampleInst = new DataGenCPInstruction(null, Hop.DataGenMethod.SAMPLE, null,
+				new CPOperand("sample", Expression.ValueType.DOUBLE, Expression.DataType.MATRIX),
+				new CPOperand(String.valueOf(mo.getNumRows()), Expression.ValueType.INT, Expression.DataType.SCALAR,
+						true), new CPOperand("1", Expression.ValueType.INT, Expression.DataType.SCALAR, true),
```

Comment:
You may directly call `MatrixBlock.sampleOperations` and `ternaryOperations` for sample and table.